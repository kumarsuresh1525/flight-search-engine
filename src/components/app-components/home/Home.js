import React, {memo} from 'react';
import './Home.scss';

const Home = ({flights = [], filters}) => {
  const journey = `${filters.src} > ${filters.dest} ${filters.returnDate ? ` > ${filters.src}` : ''}`
  return (
    <div className='main-content'>
      <div className='result-container'>
        {
          !flights.length && filters.src && <div className='default-message'>
            <p>Sorry no result Found!</p>
            <p>Please refine your search</p>
          </div>
        }
        {!flights.length && !filters.src  && <div className='default-message'>
          <p>Welcome!</p>
          <p>Please use search side bar for the customized search.</p>
          <p>Under Round Trip tab you can search for Round Trips.</p>
          <p>Please use Filter by Price range slider to refine your search.</p>
          <p>As I've created a directive to disable previous dates,</p>
          <p>So for Convenience please use below dates to test this application:</p>
          <hr />
          <div>
            <p>Data is mostly created for flights 27 Feb 2010 and Return Date: 29 Feb 2020</p>
          </div>
        </div>}
      </div>
      {flights.length > 0 &&
        <div className='flights'>
          <div className='journey'>
            <span>{journey}</span>
            <div>
              <div>Start Date: {filters.startDate}</div>
              {filters.returnDate && <div>Return Date: {filters.returnDate}</div>}
            </div>
          </div>
          {flights.map((flight) => {
            return (
              <div className='flight' key={flight.id}>
                <div className='details'>
                  <div className='flight-price'>Rs. {flight.fare}</div>
                  <div className='more-info'>
                    <div>
                      <div>{flight.flightNo}</div>
                      <div>{flight.source} > {flight.dest}</div>
                      <div>Depart: {flight.time.depart}</div>
                      <div>Arrive: {flight.time.arrive}</div>
                    </div>
                    {
                      filters.returnDate && <div>
                        <div>{flight.round_trip.number}</div>
                        <div>{flight.dest} > {flight.source}</div>
                        <div>Depart: {flight.round_trip.depart}</div>
                        <div>Arrive: {flight.round_trip.arrive}</div>
                      </div>
                    }
                  </div>
                </div>
                <div className='book'>
                  <img src={flight.imgUrl} alt='flight icon'/>
                  <button className='btn btn-primary'>Book</button>
                </div>
              </div>
            );
          })}
        </div>
      }
    </div>
  );
}

export default memo(Home);