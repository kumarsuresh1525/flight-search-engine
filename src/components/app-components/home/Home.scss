@import '../../../styles/variables';

.main-content {
  display: inline-block;
  max-width: 75%;
  width: 100%;
  height: calc(100% - 60px);
  vertical-align: top;

  .result-container {
    padding: 0.6rem;

    .default-message {
      color: $indigo;
      font-size: 18px;
      font-weight: 400;
      line-height: 1.5;
      margin-top: 3.25em;
      text-align: center;

      > div {
        font-weight: bold;
      }
    }
  }

  .flights {
    padding: 0;
    .journey {
      background-color: $red;
      padding: 20px;
      margin-bottom: 1em;
      color: $white;
      font-size: 16px;

      div:last-child {
        float: right;
        display: grid;
      }
    }

    .flight {
      box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
      margin-bottom: 0.625em;
      overflow: hidden;
      padding: 1em;
      transition: all 0.3s cubic-bezier(.25, .8, .25, 1);

      .details {
        float: left;

        .more-info {
          margin-top: 10px;
          display: flex;

          > div {
            font-size: 17px;
            > div {
              margin-bottom: 10px;
            }
            &:first-child {
              margin-right: 150px;
            }
          }
        }

        .flight-price {
          font-weight: bold;
        }
      }

      .book {
        float: right;
        max-width: 20%;
        text-align: center;

        > img {
          width: 100px;
          mix-blend-mode: multiply;
        }

        .btn-primary {
          min-width: 7em;
          background-color: $red;
          box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.12), 0 1px 5px 0 rgba(0, 0, 0, 0.2);
          border-radius: 2px;
          cursor: pointer;
          padding: 10px 15px;
        }
      }

      &:hover {
        background-color: rgba(0, 0, 0, 0.1);
      }
    }
  }
}

@media screen and (max-width: 768px) {
  .main-content,
  .result-container {
    max-width: 100%;
    border: 0;
    height: auto;
  }
  .main-content {
    border-bottom: 5px solid #ccc;
  }
}