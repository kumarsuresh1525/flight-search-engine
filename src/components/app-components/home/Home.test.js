import React from 'react';
import { render } from '@testing-library/react';
import Home from './Home';

describe('Home Page test cases', () => {
  const defaultProps = {
    flights: [],
    filters: {
      src: 'src',
      dest: 'dest'
    }
  }

  const flights = [
    { id: 1,
      flightNo: "EM-P-D-201",
      source: "PNQ",
      source_name: "Pune",
      dest: "DEL",
      dest_name: "Delhi",
      fare: "2500.00",
      date: "2020-02-27",
      time: {depart: "08:00 AM", arrive: "10:30 AM"},
      round_trip_date: "2020-02-29",
      round_trip: {depart: "07:00 AM", arrive: "10:30 PM", number: "EM-D-P-201"},
      imgUrl: 'imgUrl'
    }
  ]

  test('renders Home correctly when list is empty', () => {
    const { asFragment } = render(<Home {...defaultProps}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders Home correctly with flights details', () => {
    const props = {...defaultProps, flights: flights};
    const { asFragment } = render(<Home {...props}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should be able to no result found', () => {
    const props = {...defaultProps, filters: {returnDate: '2020-02-29', src: 'src', dest: 'dest', startDate: '2020-02-27'}}
    const { getByText } = render(<Home {...props}/>);
    expect(getByText(/Sorry no result Found!/)).toBeInTheDocument();
  });

  it('Should be able to show results for round trip', () => {
    const flights = [
      { id: 1,
        flightNo: "EM-P-D-201",
        source: "PNQ",
        source_name: "Pune",
        dest: "DEL",
        dest_name: "Delhi",
        fare: "2500.00",
        date: "2020-02-27",
        time: {depart: "08:00 AM", arrive: "10:30 AM"},
        round_trip_date: "2020-02-29",
        round_trip: {depart: "07:00 AM", arrive: "10:30 PM", number: "EM-D-P-201"},
        imgUrl: 'imgUrl'
      }
    ]
    const props = {flights, filters: {returnDate: '2020-02-29', src: 'src', dest: 'dest', startDate: '2020-02-27'}}
    const { getByText } = render(<Home {...props}/>);
    expect(getByText(/Start Date: 2020-02-27/)).toBeInTheDocument();
    expect(getByText(/Return Date: 2020-02-29/)).toBeInTheDocument();
  });
});

