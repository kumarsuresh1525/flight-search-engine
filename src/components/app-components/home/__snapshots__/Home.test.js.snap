// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page test cases renders Home correctly when list is empty 1`] = `
<DocumentFragment>
  <div
    class="main-content"
  >
    <div
      class="result-container"
    >
      <div
        class="default-message"
      >
        <p>
          Sorry no result Found!
        </p>
        <p>
          Please refine your search
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home Page test cases renders Home correctly with flights details 1`] = `
<DocumentFragment>
  <div
    class="main-content"
  >
    <div
      class="result-container"
    />
    <div
      class="flights"
    >
      <div
        class="journey"
      >
        <span>
          src &gt; dest 
        </span>
        <div>
          <div>
            Start Date: 
          </div>
        </div>
      </div>
      <div
        class="flight"
      >
        <div
          class="details"
        >
          <div
            class="flight-price"
          >
            Rs. 2500.00
          </div>
          <div
            class="more-info"
          >
            <div>
              <div>
                EM-P-D-201
              </div>
              <div>
                PNQ &gt; DEL
              </div>
              <div>
                Depart: 08:00 AM
              </div>
              <div>
                Arrive: 10:30 AM
              </div>
            </div>
          </div>
        </div>
        <div
          class="book"
        >
          <img
            alt="flight icon"
            src="imgUrl"
          />
          <button
            class="btn btn-primary"
          >
            Book
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
